backend/src/apiserver/resource/resource_manager.go

```diff
@@ -1152,3 +1154,161 @@ func (r *ResourceManager) getNamespaceFromExperiment(references []*api.ResourceR
+	workflow.SetAnnotations(common.ArtifactEndpointAnnotation, common.GetArtifactEndpoint())
+	workflow.SetAnnotations(common.ArtifactEndpointSchemeAnnotation, common.GetArtifactEndpointScheme())
+
+	// Process artifacts
+	artifactItems, exists := workflow.ObjectMeta.Annotations[common.ArtifactItemsAnnotation]
+
+	// Only inject artifacts if the necessary annotations are provided.
+	if exists {
+		var artifactItemsJSON map[string][][]interface{}
+		if err := json.Unmarshal([]byte(artifactItems), &artifactItemsJSON); err != nil {
+			panic(err)
+		}
+		for _, task := range workflow.Spec.PipelineSpec.Tasks {
+			artifacts, hasArtifacts := artifactItemsJSON[task.Name]
+			archiveLogs := common.IsArchiveLogs()
+			trackArtifacts := common.IsTrackArtifacts()
+			stripEOF := common.IsStripEOF()
+			if (hasArtifacts && len(artifacts) > 0 && trackArtifacts) || archiveLogs || (hasArtifacts && len(artifacts) > 0 && stripEOF) {
+				// Need to represent as Raw String Literals
+				artifactScript := common.GetArtifactScript() + "\n"
+				if archiveLogs {
+					loggingScript := "cat /var/log/containers/$PODNAME*step-main*.log > step-main.log && " +
+						"push_artifact main-log step-main.log\n"
+					artifactScript += loggingScript
+
+					// Logging volumes
+					if task.TaskSpec.Volumes == nil {
+						task.TaskSpec.Volumes = []corev1.Volume{}
+					}
+					loggingVolumes := []corev1.Volume{
+						r.getHostPathVolumeSource("varlog", "/var/log"),
+						r.getHostPathVolumeSource("varlibdockercontainers", "/var/lib/docker/containers"),
+						r.getHostPathVolumeSource("varlibkubeletpods", "/var/lib/kubelet/pods"),
+						r.getHostPathVolumeSource("varlogpods", "/var/log/pods"),
+					}
+					task.TaskSpec.Volumes = append(task.TaskSpec.Volumes, loggingVolumes...)
+
+					// Logging volumeMounts
+					if task.TaskSpec.StepTemplate == nil {
+						task.TaskSpec.StepTemplate = &corev1.Container{}
+					}
+					if task.TaskSpec.StepTemplate.VolumeMounts == nil {
+						task.TaskSpec.StepTemplate.VolumeMounts = []corev1.VolumeMount{}
+					}
+					loggingVolumeMounts := []corev1.VolumeMount{
+						{
+							Name:      "varlog",
+							MountPath: "/var/log",
+						},
+						{
+							Name:      "varlibdockercontainers",
+							MountPath: "/var/lib/docker/containers",
+							ReadOnly:  true,
+						},
+						{
+							Name:      "varlibkubeletpods",
+							MountPath: "/var/lib/kubelet/pods",
+							ReadOnly:  true,
+						},
+						{
+							Name:      "varlogpods",
+							MountPath: "/var/log/pods",
+							ReadOnly:  true,
+						},
+					}
+					task.TaskSpec.StepTemplate.VolumeMounts = append(task.TaskSpec.StepTemplate.VolumeMounts, loggingVolumeMounts...)
+				}
+
+				// Upload Artifacts if the artifact is enabled and the annoations are present
+				if hasArtifacts && len(artifacts) > 0 && trackArtifacts {
+					for _, artifact := range artifacts {
+						if len(artifact) == 2 {
```

Comment:
It does nothing if the array doesn't have two elements. Shouldn't it raise an error or at least log a warning?

Reply:
Thanks, I added a warning if this condition fails.